app/models/game.rb

```diff
@@ -50,7 +50,16 @@ def get_player_two
 
   def check?
     kings = pieces.where(type: 'King')
-    kings.any? { |king| pieces.any? { |piece| piece.can_take?(king) } }
+    kings.each do |king|
+      if king.piece_number > 5 && pieces.any? { |piece| piece.can_take?(king) }
+        self.state = 'Black King in Check'
+        return true
+      elsif king.piece_number < 6 && pieces.any? { |piece| piece.can_take?(king) }
```

Comment:
We should be able to use is_white? here since that function sits in the superclass.

Reply:
Very true. Will fix when I get home.